dockers/docker-orchagent/ipinip_ping.py

```diff
@@ -0,0 +1,197 @@
+                return None
+            logger.log_info("Not all portchannel members are up")
+            time.sleep(interval)
+
+        raise RuntimeError('Portchannel member interfaces did not come '
+                           'up within {}'.format(timeout))
+
+    def get_ipinip_tunnel_addrs(self):
+        """
+        Get the IP addresses used for the IPinIP tunnel
+
+        These should be the Loopback0 addresses for this device and the
+        peer device
+
+        Returns:
+            ((str) self_loopback_ip, (str) peer_loopback_ip)
+            or
+            (None, None) if an error is encountered. This most likely means
+                        the host device is not a dual ToR device
+        """
+        try:
+            peer_switch = self.config_db.get_keys('PEER_SWITCH')[0]
+            tunnel = self.config_db.get_keys('TUNNEL')[0]
+        except IndexError:
+            logger.log_warning('PEER_SWITCH or TUNNEL table'
+                               'not found in config DB')
+            return None, None
+
+        try:
+            peer_loopback_ip = self.config_db.get_entry(
+                                    'PEER_SWITCH', peer_switch
+                                    )['address_ipv4']
+            self_loopback_ip = self.config_db.get_entry(
+                                    'TUNNEL', tunnel
+                                    )['dst_ip']
+        except KeyError as e:
+            logger.log_warning(
+                'PEER_SWITCH or TUNNEL table missing data, '
+                'could not find key {}'
+                .format(e)
+            )
+            return None, None
+
+        return self_loopback_ip, peer_loopback_ip
+
+    def get_inner_pkt_type(self, packet):
+        """
+        Get the type of an inner encapsulated packet
+
+        Returns:
+            (str)  'v4' if the inner packet is IPv4
+            (str)  'v6' if the inner packet is IPv6
+            (bool) False if `packet` is not an IPinIP packet
+        """
+        if packet.haslayer(IP):
+            # Determine inner packet type based on IP protocol number
+            # The outer packet type should always be IPv4
+            if packet[IP].proto == 4:
+                return IP
+            elif packet[IP].proto == 41:
+                return IPv6
+        return False
+
+    def listen_for_ipinip(self):
+        """
+        Listens for IPinIP packets that are trapped to CPU
+
+        These packets may be trapped if there is no neighbor info for the
+        inner packet destination IP in the hardware.
+        """
+
+        def _ping_inner_dst(packet):
+            """
+            Pings the inner destination IP for an IPinIP packet
+
+            Args:
+                packet: The IPinIP packet received
+            """
+            inner_packet_type = self.get_inner_pkt_type(packet)
+            if inner_packet_type and packet[IP].dst == self_ip:
+                cmds = ['ping', '-c1', '-W1']
```

Comment:
please use [timeout ](https://github.com/Azure/sonic-buildimage/blob/master/files/scripts/arp_update#L68)command so that the command may exit gracefully.